test/TestExtensions/API.Test/InternalAPITestHook.cs

```diff
@@ -152,5 +152,15 @@ public static bool BatchEventsApi(string id, string version)
                            batchStartIds[0].Equals(batchEndIds[0], StringComparison.Ordinal);
                 });
         }
+
+        public static IVsProjectJsonToPackageReferenceMigrateResult MigrateJsonProject(string projectName)
+        {
+            var migrator = ServiceLocator.GetComponent<IVsProjectJsonToPackageReferenceMigrator>();
+            return ThreadHelper.JoinableTaskFactory.Run(async () =>
+            {
+                await ThreadHelper.JoinableTaskFactory.SwitchToMainThreadAsync();
```

Comment:
no you are right, the switching is done by the api itself so we don't need to do it here. have fixed it.